### PR TITLE
rafthttp/metrics.go:fixed TODO: record write/recv failures.

### PR DESCRIPTION
**Pull request by vimalk78**
added two counters in rafthttp/metrics.go

-   sendFailures
-   recvFailures

### CODE REVIEW

**Comment by xiang90** [comment]
This approach looks good to me. I need to double check if there is any other place that send/receive can fail.

File: Documentation/metrics.md [hunk 1]
```diff
@@ -70,6 +70,8 @@ All these metrics are prefixed with `etcd_network_`
 |---------------------------|--------------------------------------------------------------------|---------------|
 | peer_sent_bytes_total           | The total number of bytes sent to the peer with ID `To`.         | Counter(To)   |
 | peer_received_bytes_total       | The total number of bytes received from the peer with ID `From`. | Counter(From) |
+| peer_sent_faulures_total        | The total number of send failures to the peer with ID `To`.         | Counter(To)   |
```
Comment (xiang90) [inline Documentation/metrics.md]: faulures -> failures

File: rafthttp/metrics.go [hunk 2]
```diff
@@ -36,6 +35,24 @@ var (
 		[]string{"From"},
 	)
 
+	sendFailures = prometheus.NewCounterVec(prometheus.CounterOpts{
+		Namespace: "etcd",
+		Subsystem: "network",
+		Name:      "peer_send_failures_total",
```
Comment (xiang90) [inline rafthttp/metrics.go]: send -> sent

File: Documentation/metrics.md (hunk 1 above)
Comment (xiang90) [inline Documentation/metrics.md]: @heyitsanthony should this be `The total number of sending failures`? or sent failures?

**Comment by xiang90** [comment]
@heyitsanthony Can you help to double check the grammar of the metrics descriptions? Thanks!

File: Documentation/metrics.md (hunk 1 above)
Comment (heyitsanthony) [inline Documentation/metrics.md]: `number of send failures from the peer`

File: rafthttp/metrics.go [hunk 3]
```diff
@@ -36,6 +35,24 @@ var (
...
+		Help:      "The total number of send failures to peers.",
+	},
+		[]string{"To"},
+	)
+
+	recvFailures = prometheus.NewCounterVec(prometheus.CounterOpts{
+		Namespace: "etcd",
+		Subsystem: "network",
+		Name:      "peer_recv_failures_total",
```
Comment (heyitsanthony) [inline rafthttp/metrics.go]: s/recv/receive to be consistent with `peer_received_bytes_total`?

**Comment by xiang90** [comment]
LGTM. Can you please squash the commits into one?  Then we can get this merged.

**Comment by vimalk78** [comment]
@xiang90 i have squashed the commits.

File: Documentation/libraries-and-tools.md [hunk 4]
```diff
@@ -23,6 +23,7 @@
 
 **Java libraries**
 
+- [coreos/jetcd](https://github.com/coreos/jetcd) - Supports v3
```
Comment (xiang90) [inline Documentation/libraries-and-tools.md]: can we remove this line? this is irrelevant with this pr.

**Comment by xiang90** [comment]
lgtm. defer to @heyitsanthony

File: Documentation/libraries-and-tools.md (hunk 4 above)
Comment (vimalk78) [inline Documentation/libraries-and-tools.md]: done.

**Comment by heyitsanthony** [comment]
lgtm following minor string fixup. Thanks!

**Pull request merged**